docs/continuous-delivery/gitops/secret-management/secret-management-gitops.md

```diff
@@ -0,0 +1,189 @@
+# Secret Management with GitOps
+
+GitOps projects have git repositories in the center that are considered a source of truth for managing both infrastructure and application. This infrastructure and application will require secured access to other resources of the system through secrets. Committing clear-text secrets into git repositories is unacceptable even if the repositories are private to your team and organization. This is unacceptable for a multitude of reasons, but an example can be because secrets are then stored in clear-text on every developers computer and if a developers computer is compromised, so are all the secrets. Teams need a secure way to handle secrets when using GitOps.
+
+There are many ways to manage secrets with GitOps and at high level can be categorized into:
+
+1. Encrypted secrets in git repositories
+1. Reference to secrets stored in the external key vault
+
+> **TLDR**: Referencing secrets in an external key vault is the recommended approach, because of, but not limited to, storing encrypted secrets in repo can result in secrets being exposed if the private key is leaked, easier to orchestrate secret rotation and not as scalable to with multiple clusters or teams.
+
+## 1. Encrypted Secrets in Git Repositories
+
+In this approach, secrets are manually encrypted by developers using a public key and the custom Kubernetes controller running in the target cluster can only decrypt the key. Some popular tools for his approach are [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets), [Mozilla SOPS](https://github.com/mozilla/sops)
+
+All the secret encryption tools share the following:
+
+- Secret changes are managed by making changes within the GitOps repository which provides great traceability of the changes made
+- All secrets can be rotated by making changes in GitOps, without accessing the cluster
+- Secrets are stored encrypted in the gitops repository, if the private encryption key is leaked, all secrets can be decrypted
```

Comment:
This sounds scary

Reply:
This is what it means to have encrypted secrets in the repository and is one the downsides of course.